### DISCUSSION TITLE
Support FreeDesktop Common Extended Attributes.

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [community/community#57080] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

As discussed at [KDE's Discourse instance](https://discuss.kde.org/t/markdown-processing-in-dolphin-from-inode-properties/1718/9?u=rokejulianlockhart#:~:text=Right%2C%20I%20don,on%20the%20system.):

<blockquote>
<blockquote>
<blockquote>
About the tag functionality, you might be interested to know that there is already a cross-desktop way to tag files. <code>user.xdg.tags</code> in the extended attributes is typically used for this AFAIK. This isn’t Dolphin-specific or anything, and there is technically nothing stopping anyone from supporting this everywhere.
</blockquote>

Does an equivalent exist for comments too?
</blockquote>

Yes! This is actually one of the few attributes that are mentioned on [FreeDesktop.org](https://www.freedesktop.org/wiki/CommonExtendedAttributes/#:~:text=the%20original%20message.-,These%20attributes%20are%20currently,Uid%3A%20Used%20by%20Beagle,-Proposed%20control%20attributes) as [Common Extended Attributes](https://wiki.archlinux.org/index.php?title=Extended_attributes&oldid=798672#User_extended_attributes:~:text=Some%20other%20user,for%20each%20file.): <code>user.xdg.comment</code>.
</blockquote>

Most common extended filesystem attributes are standardized, and supported by most other filesystems. However, *GitHub doesn't appear to retain file comments when uploaded.*

Correct me if I am incorrect, please. If this is verified, I'll report at https://support.github.com/contact/bug-report.

**Comment by airtower-luna** [comment]
I think you're asking for support for extended attributes in Git? That'd be a feature request for *Git*, not GitHub. :thinking: 

In that case you should look here: https://www.git-scm.com/community

**Reply by roke-julian-lockhart** [reply]
Oh, does `git` not support it, @airtower-luna?

**Reply by airtower-luna** [reply, marked as answer]
The extent to which Git stores file metadata (other than the filename) is about this (from [Git Internals - Git Objects](https://git-scm.com/book/en/v2/Git-Internals-Git-Objects)):

> In this case, you’re specifying a mode of `100644`, which means it’s a normal file. Other options are `100755`, which means it’s an executable file; and `120000`, which specifies a symbolic link. The mode is taken from normal UNIX modes but is much less flexible — these three modes are the only ones that are valid for files (blobs) in Git (although other modes are used for directories and submodules).

And searching the source code doesn't turn up anything either, so I'm very sure.

**Reply by roke-julian-lockhart** [reply]
Indeed – per https://stackoverflow.com/a/2345786/9731176, it definitely doesn't, although stopgap solutions exist.
